docs/cvl/cvl2/changes.md

```diff
@@ -0,0 +1,586 @@
+Syntax changes introduced in CVL 2
+==================================
+
+CVL 2.0 is a major overhaul to the type system of CVL.  Many
+of the changes are internal, but we also wanted to take this opportunity to
+introduce a few improvements to the syntax.  The general goal of these changes
+is to make the behavior of CVL more explicit and predictable, and to bring the
+syntax more in line with Solidity's syntax.
+
+This document summarizes the changes to CVL syntax introduced by CVL 2.0.
+
+```{contents}
+```
+
+Superficial syntax changes
+--------------------------
+
+There are several simple changes to the syntax to make specs more uniform and
+consistent, and to reduce the superficial differences with Solidity.
+
+### `function` and `;` required for methods block entries
+Methods block entries must now start with `function` and end with `;`.  For
+example:
+
+```cvl
+balanceOf(address) returns(uint) envfree
+```
+will become
+```cvl
+function balanceOf(address) external returns(uint) envfree;
+```
+(note also the addition of `external`, see {ref}`described below <cvl2-visibility>`).
+
+This is also true for entries with summaries:
+```cvl
+_setManagedBalance(address,uint256) => NONDET
+```
+will become
+```cvl
+function _setManagedBalance(address,uint256) internal => NONDET;
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Required `;` in more places
+
+`using`, `pragma`, `import`, and `use` statements all require a `;` at the end.  For
+example,
+
+```cvl
+using C as c
+```
+
+becomes
+```cvl
+using C as c;
+```
+
+Note: `use` statements do not require (and may not have) a semicolon if they
+are followed by a `preserved` or `filtered` block.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Method literals require `sig:`
+
+In some places in CVL, you can refer to a contract method by its name and
+argument types.  For example, you might write
+```cvl
+require f.selector == balanceOf(address).selector;
+```
+
+In this example, `balanceOf(address)` is a *method literal*.  In CVL 2,
+these methods literals must now start with `sig:`.  For example, the above
+would become:
+
+```cvl
+require f.selector == sig:balanceOf(address).selector;
+```
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Stricter ordering on method annotations
+
+In CVL 2, the order of the annotations must be visibility modifiers (`internal` or `external`),
+followed by `returns` clause (if any), followed by `optional`, `library`, or `envfree` in any order (if any),
+followed by a summary (if any).
+
+CVL 1 was less strict about the order.
+
+```{todo}
+If you do not change this, you will see the following error:
+```
+
+### Use of contract name instead of `using` variable
+
+In CVL 1, the only way to refer to a contract on the scene was to first
```

Comment:
I think "on the scene" is a technical term here rather than just a usage of the normal English colloquialism, yes? I was initially confused, although that may be my lack of familiarity with this usage.

Reply:
added a link to the glossary